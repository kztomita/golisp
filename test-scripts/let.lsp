(setq outer_var 100)
(setq b 10)
(setq y 20)

(defun foo (a b)
  (let ((x 100) (y 200))
    (print (list outer_var a b x y))
    (+ outer_var a b x y)))

(foo 2 3)

(print (list outer_var b y))

(print "====")
(let ()
  (print "nil binding list"))
(let nil
  (print "nil binding list"))

