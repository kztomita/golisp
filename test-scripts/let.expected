
(100 2 3 100 200) 
(100 10 20) 
"====" 
"nil binding list" 
"nil binding list" 
